databases/week3/mongo.txt

```diff
@@ -0,0 +1,213 @@
+        "contact_email": "email@gmail.com"
+    },
+
+    {
+        "_id": 2,
+        "number_of_guests": 6,
+        "meal_id": 2,
+        "created_date": null,
+        "contact_phonenumber": 77777777,
+        "contact_name": "Kolya",
+        "contact_email": null
+    }
+])
+
+db.Review.insertMany([
+    {
+        "_id": 1,
+        "review_titl": "Argentinas meal",
+        "description": "I loved the asado on the patio, and the view from Nicolas' apartment is amazing. It's one of my favorite memories from my time in Buenos Aires! Nicolas is an easy-going, friendly cook, and his apartment is the perfect date night location. Nicolas was warm and welcoming from beginning to end, and that feeling of simple goodness carries into the food. It would be easy to spend a long time chatting on the patio, except that he has so many recommendations of what to do next in the area (ask about PIBÄ).",
+        "meal_id": 1,
+        "stars": 5,
+        "created_date": ISODate("2020-05-18T14:10:30.000Z")  
+    },
+
+    {
+        "_id": 2,
+        "review_titl": "Japanese meal",
+        "description": "I booked a cooking class with Furu through Traveling Spoon. He offers a variety of dishes to learn. I was most curious about Tamago-yaki, the Japanese omelette. Furu was a great teacher; rolling and flipping the Tamago-yaki is much more difficult than it appears! We also prepared Japanese hot pot and soba for the rest of the meal. Great experience and delicious food!",
+        "meal_id": 2,
+        "stars": 7,
+        "created_date": ISODate("2021-01-15T06:31:15.000Z")
+    }
+])
+
+------------------------------------------------------------------------- Get with any id, fx 1
+
+db.Meal.find({_id: 1})
+db.Reservation.find({_id: 1})
+db.Review.find({_id: 1})
+
+-----------------------------------------------------------------------------Update with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+db.Meal.updateOne({_id: 1}, {$set: {date: new ISODate("2022-05-18T14:10:30.000Z")}})
+db.Reservation.updateOne({_id: 1}, {$inc: {number_of_guests: 1}})
+db.Review.updateOne({_id: 2}, {$inc: {stars: -1}})
+
+--------------------------------------------------------------------- Delete with any id, fx 1
+
+db.Meal.deleteOne({_id: 1})
+db.Reservation.deleteOne({_id: 1})
+db.Review.deleteOne({_id: 1})
+
+
+---------------------------------------------------------------- Get meals that has a price smaller than a specific price fx 90
+
+db.Meal.find({price: {$lt: 50}})
+
+----------------------------------------------------------------Get meals that still has available reservations
+
+db.Meal.updateMany({}, {$set: {status: "available"}})
+db.Meal.find({status: { $eq: "available"}})
+
+------------------------------------------------------------------ Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+
+db.Meal.find( { meal_titl: { $in: [ /^Sushi/] } } )
+
```

Comment:
Also you can write query like this:
db.Meal.find({ meal_titl: { $regex: /sushi/i } }) - this means "sushi" can be found anywhere in the string and it's case-insensitive.